closure/goog/testing/asserts_test.js

```diff
@@ -436,9 +438,61 @@ function testAssertObjectEqualsMap() {
   assertObjectEquals('maps should be equal', map1, map2);
 
   map1.set('hi', 'hey');
-  assertThrows('sets should not be equal', function() {
-    assertObjectEquals(map1, map2);
-  });
+  assertThrows('maps should not be equal',
+               function() { assertObjectEquals(map1, map2); });
+}
+
+
+function testAssertObjectEqualsTypedArray() {
+  var typedArrays = [
+    'Float32Array',
+    'Float64Array',
+    'Int8Array',
+    'Int16Array',
+    'Int32Array',
+    'Uint8Array',
+    'Uint8ClampedArray',
+    'Uint16Array',
+    'Uint32Array',
+    'BigInt64Array',
+    'BigUint64Array',
+  ];
+
+  var prior, priorName;
+  for (var i = 0; i < typedArrays.length; ++i) {
+    var name = typedArrays[i];
+    var arrType = window[name];
+    if (arrType) {
+      function construct(num) { return num; }
+      function mutate(array, offset) { array[0] *= -1; }
+      if (name.startsWith('Big')) {
```

Comment:
I'm not in love with all this logic in a test case.  I wonder if simply breaking out a few relevant cases wouldn't be cleaner?

1. same elements in same type of array with different size backing array is equal
2. do a mutation to one of the normal TypedArrays and to one of the BigInt ones and make sure it makes them unequal
3. a simple pair of each are equal (this one should be exhaustive but shouldn't need a for-loop - e.g. "assertObjectEquals(Uint8Array.of(1, 2, 3), Uint8Array.of(1, 2, 3)); assertObjectEquals(Uint16Array.of(1, 2, 3), Uint16Array.of(1, 2, 3));"
4. same elements in different type are unequal (again, just pick a couple pairs)
5. different elements same length => unequal
6. same elements but one extra => unequal

For the bigint ones, we can jam them into a single separate test case where the whole thing is gated on an existence check.

Reply:
I've done this, which makes all the above review changes irrelevant.  Thanks for helping my write better unit tests!